homeassistant/components/binary_sensor/ffmpeg.py

```diff
@@ -73,8 +77,16 @@
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Create the binary sensor."""
-    from haffmpeg import SensorNoise, SensorMotion
+    from haffmpeg import Test, SensorNoise, SensorMotion
+
+    # check source
+    if config.get(CONF_RUN_TEST):
+        test = Test(config.get(CONF_FFMPEG_BIN))
+        if not test.run_test(config.get(CONF_INPUT)):
+            _LOGGER.critical("FFmpeg test fails!")
```

Comment:
I would consider this to be an error, not critical, as it doesn't take the whole system down